lightning/src/ln/channelmanager.rs

```diff
@@ -2426,6 +2484,11 @@ where
 	/// irrevocably committed to on our end. In such a case, do NOT retry the payment with a
 	/// different route unless you intend to pay twice!
 	///
+	/// Thus, in order to ensure duplicate payments are not sent, if we begin the process of sending
+	/// a payment, but crash before `send_payment` returns (or prior to [`ChannelMonitorUpdate`]
+	/// persistence if you're using [`ChannelMonitorUpdateStatus::InProgress`]), the payment may be
+	/// lost on restart. See [`ChannelManager::list_recent_payments`] for more information.
+	///
```

Comment:
Err.. sorry, what I meant was what you wrote in the previous commit belongs as part of the paragraph starting on line 2461. That paragraph says, "you should implement your own tracking of payments." So it seems like a natural place to talk about `list_recent_payments`.